restapi/src/main/java/io/stargate/web/resources/Db.java

```diff
@@ -100,7 +103,11 @@ private DataStore getDataStoreInternal(
             .build();
 
     DataStoreOptions options =
-        DataStoreOptions.builder().defaultParameters(parameters).alwaysPrepareQueries(true).build();
+        DataStoreOptions.builder()
+            .defaultParameters(parameters)
+            .alwaysPrepareQueries(true)
+            .customProperties(authenticationSubject.customProperties())
```

Comment:
This looks a bit inconsistent with the other places in restapi. In other cases we use full HTTP headers, but here only the subject's headers, which is not guaranteed to be the same. One level up the call chain to this method full headers are available... perhaps we could just pass them here as one extra parameter?

As for `AuthenticationSubject`, would it make sense to _combine_ subject properties (may be set by auth services) with the raw headers? The subject's properties are different from raw HTTP headers, but will reflect the authenticators interpretation of the request, which might be worth examining by persistence implementations. WDYT?

Reply:
Ideally these would be the raw http headers, I didn't see where those were.

Reply:
Would you mind pushing that change?

Reply:
sure

Reply:
implemented that in c16c71458ee7989a06e0222898978775f6dac415